image build fails without resteasy using quarkus-openshift extension

**Describe the bug**
When trying to build with Openshift extension with the following command:
./mvnw clean package -Dquarkus.container-image.build=true

**Expected behavior**
Build should be succesfull

**Actual behavior**
It seems that the openshift.yml and other kubernetes related files are not generated. The target/kubernetes directory is empty

following error is thrown

```
[ERROR] Failed to execute goal io.quarkus:quarkus-maven-plugin:1.3.2.Final:build (default) on project test-without-resteasy-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR]         [error]: Build step io.quarkus.container.image.s2i.deployment.S2iProcessor#s2iBuildFromJar threw an exception: java.lang.IllegalStateException: Could not find kubernetes/openshift.yml
[ERROR]         at io.quarkus.container.image.s2i.deployment.S2iProcessor.lambda$s2iBuildFromJar$5(S2iProcessor.java:185)
```


**To Reproduce**
Steps to reproduce the behavior:
1. ./mvnw clean package -Dquarkus.container-image.build=true
2. Build will fail
3. Add the resteasy dependency and the build will work.

Following project attached to reproduce
[image-build-failed.tar.gz](https://github.com/quarkusio/quarkus/files/4452886/image-build-failed.tar.gz)

**Configuration**
```properties
# Add your application.properties here, if applicable.
quarkus.kubernetes-client.trust-certs=true
```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
Linux localhost 5.5.15-200.fc31.x86_64 #1 SMP Thu Apr 2 19:16:17 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

 - Output of `java -version`: 
openjdk version "11.0.6" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.6+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.6+10, mixed mode)

 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 
1.3.2.Final

 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /home/.m2/wrapper/dists/apache-maven-3.6.3-bin/1iopthnavndlasol9gbrbg6bf2/apache-maven-3.6.3
Java version: 11.0.6, vendor: AdoptOpenJDK, runtime: /home/jdk-11.0.6+10
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "5.5.15-200.fc31.x86_64", arch: "amd64", family: "unix"

**Additional context**
(Add any other context about the problem here.)

--- a/extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java
+++ b/extensions/container-image/container-image-s2i/deployment/src/main/java/io/quarkus/container/image/s2i/deployment/S2iProcessor.java
@@ -174,17 +174,23 @@ public class S2iProcessor {
             return;
         }
 
+        Optional<GeneratedFileSystemResourceBuildItem> openshiftYml = generatedResources
+                .stream()
+                .filter(r -> r.getName().endsWith("kubernetes/openshift.yml"))
+                .findFirst();
+
+        if (!openshiftYml.isPresent()) {
+            LOG.warn(
+                    "No Openshift manifests were generated (most likely due to the fact that the service is not an HTTP service) so no s2i process will be taking place");
+            return;
+        }
+
         String namespace = Optional.ofNullable(kubernetesClient.getClient().getNamespace()).orElse("default");
         LOG.info("Performing s2i binary build with jar on server: " + kubernetesClient.getClient().getMasterUrl()
                 + " in namespace:" + namespace + ".");
         String image = containerImage.getImage();
 
-        GeneratedFileSystemResourceBuildItem openshiftYml = generatedResources
-                .stream()
-                .filter(r -> r.getName().endsWith("kubernetes/openshift.yml"))
-                .findFirst().orElseThrow(() -> new IllegalStateException("Could not find kubernetes/openshift.yml"));
-
-        createContainerImage(kubernetesClient, openshiftYml, s2iConfig, out.getOutputDirectory(), jar.getPath(),
+        createContainerImage(kubernetesClient, openshiftYml.get(), s2iConfig, out.getOutputDirectory(), jar.getPath(),
                 out.getOutputDirectory().resolve("lib"));
         artifactResultProducer.produce(new ArtifactResultBuildItem(null, "jar-container", Collections.emptyMap()));
         containerImageResultProducer.produce(

